assets/dev/js/editor/components/template-library/manager.js

```diff
@@ -359,22 +359,29 @@ const TemplateLibraryManager = function() {
 		} );
 	};
 
-	this.getFolderTemplates = ( templateId ) => {
+	this.getFolderTemplates = ( parentElement ) => {
 		this.clearLastRemovedItems();
 
+		const parentId = parentElement.model.get( 'template_id' );
+		const parentTitle = parentElement.model.get( 'title' );
+
 		return new Promise( ( resolve ) => {
 			isLoading = true;
 			const ajaxOptions = {
 				data: {
 					source: 'cloud',
-					template_id: templateId,
+					template_id: parentId,
 				},
 				success: ( data ) => {
-					this.setFilter( 'parent', templateId );
```

Comment:
Why don't we just keep `parent`? And make it an object instead of managing multiple properties separately.